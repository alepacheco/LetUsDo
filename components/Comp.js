function Home() {
  return <div>Imported</div>;
}

export default Home;