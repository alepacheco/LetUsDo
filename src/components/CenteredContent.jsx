import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
