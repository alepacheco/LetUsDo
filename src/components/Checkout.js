import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}