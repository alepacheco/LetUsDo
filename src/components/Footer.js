import React from 'react';
import '../styles/components/Footer.scss';

export class Footer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="footer">
        <div className="separator"/>
        <div className="wrapper">
          some text here
        </div>
      </div>
    );
  }
}

export default Footer;
