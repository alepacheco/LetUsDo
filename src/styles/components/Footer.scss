.footer {
  margin-top: 120px;

  .wrapper {
    color: #999999;
    margin: 16px 0 32px 0;
    text-align: center;
    font-size: 14px;
    font-family: Quicksand, Sans-Serif;
    font-weight: 300;
  }

  .separator {
    width: 720px;
    margin-left: auto;
    margin-right: auto;
    display: block;
    height: 1px;
    border: 0;
    border-top: 1px solid #ccc;

    @media only screen and (max-width: 800px) {
      margin: 1em 38px;
      width: auto;
    }
    padding: 0;
  }
}
