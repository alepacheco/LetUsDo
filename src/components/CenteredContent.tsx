import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
