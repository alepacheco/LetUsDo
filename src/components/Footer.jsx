import React from 'react';
import '../styles/components/Footer.scss';

export class Footer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (<div className="footer" >
      <div className="separator" />
      <div className="wrapper" >
        We are in beta phase and would love to hear your feedback.
      <br />
        Please get in touch with the email: contact@letusdo.app
      </div>
    </div>
    );
  }
}

export default Footer;
