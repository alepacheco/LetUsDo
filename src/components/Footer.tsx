import React from 'react';
import 'src/styles/components/Footer.scss';

export const Footer = () => (
  <div className="footer">
    <div className="separator" />
    <div className="wrapper">
      We are in beta phase and would love to hear your feedback.
      <br />
      Please get in touch with the email: contact@letusdo.app
    </div>
  </div>
);

export default Footer;
