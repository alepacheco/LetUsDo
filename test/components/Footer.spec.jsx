import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Footer } from '../../src/components/Footer';

describe('<Footer />', () => {
  it('renders', () => {
    const wrapper = shallow(<Footer />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
