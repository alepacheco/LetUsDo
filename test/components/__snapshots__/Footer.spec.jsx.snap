// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders 1`] = `
<div
  className="footer"
>
  <div
    className="separator"
  />
  <div
    className="wrapper"
  >
    lorem ipsum dolor sit amet, consectetur adip
  </div>
</div>
`;
